cmd/kpod/pull.go

```diff
@@ -151,12 +161,34 @@ func pullCmd(c *cli.Context) error {
 		return errors.Wrapf(err, "could not create runtime")
 	}
 	for _, fqname := range fqRegistries {
+		var registryCreds *types.DockerAuthConfig
 		fmt.Printf("Trying to pull %s...", fqname)
-		if err := runtime.PullImage(fqname, c.Bool("all-tags"), c.String("signature-policy"), os.Stdout); err != nil {
+
+		reg := getRegistryOnly(fqname)
+		registryCreds, err = getCachedCreds(reg, systemContext)
+		if err != nil {
+			return errors.Wrapf(err, "error getting credentials from authfile, %q might not be logged in", reg)
+		}
+
+		options := libpod.CopyOptions{
+			SignaturePolicyPath: c.String("signature-policy"),
+			DockerRegistryOptions: common.DockerRegistryOptions{
+				DockerRegistryCreds: registryCreds,
+			},
+			AuthFile: c.String("authfile"),
+		}
+
+		if err := runtime.PullImage(fqname, c.Bool("all-tags"), options, os.Stdout); err != nil {
 			fmt.Printf(" Failed\n")
 		} else {
 			return nil
 		}
 	}
-	return errors.Errorf("error pulling image from %q", image)
+	return errors.Errorf("error pulling image from %q, might not be logged in to the registry", image)
+}
+
+// returns the registry from a path in the form "registry/image:tag"
+func getRegistryOnly(image string) string {
+	arr := strings.SplitN(image, "/", 2)
```

Comment:
How does this work with the non-`docker:` transports, documented in the man page to work?

(To be fair, this seems to be very broken already, with assuming that `alltransports.ParseImageName().DockerReference()` is never `nil`.)